src/wodan/wodan.ml

```diff
@@ -1713,18 +1676,19 @@ struct
               KeyedMap.iter (fun k off -> blit_cd_child k off entry2) cl2;
               KeyedMap.swap entry1.logdata.logdata_contents kc;
               KeyedMap.swap entry2.logdata.logdata_contents kc2;
-              entry1.children_alloc_ids <- entry.children_alloc_ids;
-              entry2.children_alloc_ids <- ca2;
+              KeyedMap.copy_in entry.children_alloc_ids
```

Comment:
I don't think copy_in is necessary (and it has a potentially large cost since fan-out is large).
The rationale is that entry.children_alloc_ids and ca2 will only be used in those places, and not shared.
This should be coordinated with _reset_contents, to make sure that entry.children_alloc_ids is replaced with a fresh instance.
Feel free to add extra comments if you feel this is overly clever (or to rename _reset_contents since it is only used here), but I think we should be careful when introducing extra costs.